## Business Understanding 

Start-up funding plays a crucial role providing essential capital to nurture new ventures that drive economic growth and technological advancement.Indian startups ecosystem span across various sectors and domains, such as e-commerce, fintech, edtech, healthtech, and agritech. This project aims to equip the team with the knowledge and strategic insights on identifying the most promising sectors, cities, funding trends, and key players necessary to make informed decisions and successfully engage with the dynamic and rapidly evolving Indian start-up landscape.

### **Data understanding**
 
The datasets contains information about startup funding from 2018 to 2021. It includes various attributes such as the company’s name, sector, funding amount, stage, investor details, and location.
 
The key attributes in the dataset include:
 
**Company/Brand**: Name of the company/start-up
 
**Founded**: Year start-up was founded
 
**Sector**: Sector of service
 
**What it does**: Description about Company
 
**Founders**: Founders of the Company
 
**Investor**: Investors
 
**Amount($)**: Raised fund
 
**Stage**: Round of funding reached
 
**Headquarters**: Location of   the start-up_company

## Hypothesis
 
Null Hypothesis (H0): Funding to start-ups is centralized around specific locations and sectors.
 
Alternative Hypothesis (H1): Funding to start-ups is spread across different locations and sectors.
 
 
 
## RESEARCH QUESTIONS
 
1.How has funding to startups changed over the period of time?
 
2.What is the average amount of funding for start-ups in?
 
3.Which headquarter is the most preferred startup location?
 
4.Which sectors are most favoured by investors?
 
5.What are the most common funding stages among indian startups?

## Install Packages

In [3]:

pip install pyodbc


Note: you may need to restart the kernel to use updated packages.


In [4]:
pip install python-dotenv 

## Import Packages Needed for This Analysis 

In [173]:
import pyodbc
#import the dotenv_values function from the dotenv package  
from dotenv import dotenv_values    
import pandas as pd
import warnings 
import numpy as np
# Ignore warning due to depreciating packages
warnings.filterwarnings('ignore')

## Loading .env file and creating a connectiong to the database

In [6]:
# Load environment variables from .env file into a dictionary
environment_variables = dotenv_values('.env')

# Get the values for the credentials you set in the '.env' file
server = environment_variables.get("server")
database = environment_variables.get("database")
username = environment_variables.get("Login")
password = environment_variables.get("password")

In [7]:
# Create a connection string
connection_string = f"DRIVER={{SQL Server}};SERVER={server};DATABASE={database};UID={username};PWD={password}"

In [8]:
# Use the pyodbc library to pass in the connection string.

connection = pyodbc.connect(connection_string)

## Accessing 2021 data from the database

In [9]:
query = "SELECT * FROM dbo.LP1_startup_funding2021"

data_2021 = pd.read_sql(query, connection)

data_2021

Company_Brand  Founded HeadQuarter                 Sector  \
0     Unbox Robotics   2019.0   Bangalore             AI startup   
1             upGrad   2015.0      Mumbai                 EdTech   
2        Lead School   2012.0      Mumbai                 EdTech   
3            Bizongo   2015.0      Mumbai         B2B E-commerce   
4           FypMoney   2021.0    Gurugram                FinTech   
...              ...      ...         ...                    ...   
1204        Gigforce   2019.0    Gurugram  Staffing & Recruiting   
1205          Vahdam   2015.0   New Delhi       Food & Beverages   
1206    Leap Finance   2019.0   Bangalore     Financial Services   
1207    CollegeDekho   2015.0    Gurugram                 EdTech   
1208          WeRize   2019.0   Bangalore     Financial Services   

                                           What_it_does  \
0     Unbox Robotics builds on-demand AI-driven ware...   
1        UpGrad is an online higher education platform.   
2     LEAD School offers technology based school tra...   
3     Bizongo is a business-to-business online marke...   
4     FypMoney is Digital NEO Bank for Teenagers, em...   
...                                                 ...   
1204                  A gig/on-demand staffing company.   
1205  VAHDAM is among the world’s first vertically i...   
1206  International education loans for high potenti...   
1207  Collegedekho.com is Student’s Partner, Friend ...   
1208  India’s first socially distributed full stack ...   

                                               Founders  \
0                           Pramod Ghadge, Shahid Memon   
1     Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                            Smita Deorah, Sumeet Mehta   
3               Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                         Kapil Banwari   
...                                                 ...   
1204                        Chirag Mittal, Anirudh Syal   
1205                                         Bala Sarda   
1206                         Arnav Kumar, Vaibhav Singh   
1207                                       Ruchir Arora   
1208                      Vishal Chopra, Himanshu Gupta   

                                           Investor        Amount  \
0                       BEENEXT, Entrepreneur First    $1,200,000   
1          Unilazer Ventures, IIFL Asset Management  $120,000,000   
2                  GSV Ventures, Westbridge Capital   $30,000,000   
3                            CDC Group, IDG Capital   $51,000,000   
4     Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000   
...                                             ...           ...   
1204                                Endiya Partners      $3000000   
1205                                       IIFL AMC     $20000000   
1206                                   Owl Ventures     $55000000   
1207               Winter Capital, ETS, Man Capital     $26000000   
1208                 3one4 Capital, Kalaari Capital      $8000000   

             Stage  
0     Pre-series A  
1             None  
2         Series D  
3         Series C  
4             Seed  
...            ...  
1204  Pre-series A  
1205      Series D  
1206      Series C  
1207      Series B  
1208      Series A  

[1209 rows x 9 columns]

## Accessing 2020 data from the database

In [10]:
query = "SELECT * FROM dbo.LP1_startup_funding2020"

data_2020 = pd.read_sql(query, connection)

data_2020

Company_Brand  Founded HeadQuarter              Sector  \
0       Aqgromalin   2019.0     Chennai            AgriTech   
1         Krayonnz   2019.0   Bangalore              EdTech   
2     PadCare Labs   2018.0        Pune  Hygiene management   
3            NCOME   2020.0   New Delhi              Escrow   
4       Gramophone   2016.0      Indore            AgriTech   
...            ...      ...         ...                 ...   
1050  Leverage Edu      NaN       Delhi              Edtech   
1051         EpiFi      NaN        None             Fintech   
1052       Purplle   2012.0      Mumbai           Cosmetics   
1053        Shuttl   2015.0       Delhi           Transport   
1054         Pando   2017.0     Chennai            Logitech   

                                           What_it_does  \
0                          Cultivating Ideas for Profit   
1     An academy-guardian-scholar centric ecosystem ...   
2      Converting bio-hazardous waste to harmless waste   
3                          Escrow-as-a-service platform   
4     Gramophone is an AgTech platform enabling acce...   
...                                                 ...   
1050  AI enabled marketplace that provides career gu...   
1051  It offers customers with a single interface fo...   
1052         Online makeup and beauty products retailer   
1053                    App based bus aggregator serice   
1054            Networked logistics management software   

                                               Founders  \
0                       Prasanna Manogaran, Bharani C L   
1                      Saurabh Dixit, Gurudutt Upadhyay   
2                                       Ajinkya Dhariya   
3                                         Ritesh Tiwari   
4     Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   
...                                                 ...   
1050                                  Akshay Chaturvedi   
1051                    Sujith Narayanan, Sumit Gwalani   
1052                          Manish Taneja, Rahul Dash   
1053                      Amit Singh, Deepanshu Malviya   
1054                     Jayakrishnan, Abhijeet Manohar   

                                   Investor      Amount       Stage column10  
0                           Angel investors    200000.0        None     None  
1                           GSF Accelerator    100000.0    Pre-seed     None  
2                            Venture Center         NaN    Pre-seed     None  
3       Venture Catalysts, PointOne Capital    400000.0        None     None  
4       Siana Capital Management, Info Edge    340000.0        None     None  
...                                     ...         ...         ...      ...  
1050  DSG Consumer Partners, Blume Ventures   1500000.0        None     None  
1051          Sequoia India, Ribbit Capital  13200000.0  Seed Round     None  
1052                             Verlinvest   8000000.0        None     None  
1053             SIG Global India Fund LLP.   8043000.0    Series C     None  
1054                      Chiratae Ventures   9000000.0    Series A     None  

[1055 rows x 10 columns]

## Accessing a csv file containing 2019 data from the root directory of this project

In [11]:

data_2019 = pd.read_csv("startup_funding2019.csv")

data_2019

Company/Brand  Founded HeadQuarter                   Sector  \
0    Bombay Shaving      NaN         NaN                Ecommerce   
1         Ruangguru   2014.0      Mumbai                   Edtech   
2          Eduisfun      NaN      Mumbai                   Edtech   
3          HomeLane   2014.0     Chennai          Interior design   
4          Nu Genes   2004.0   Telangana                 AgriTech   
..              ...      ...         ...                      ...   
84     Infra.Market      NaN      Mumbai                Infratech   
85              Oyo   2013.0    Gurugram              Hospitality   
86       GoMechanic   2016.0       Delhi  Automobile & Technology   
87           Spinny   2015.0       Delhi               Automobile   
88  Ess Kay Fincorp      NaN   Rajasthan                  Banking   

                                         What it does  \
0          Provides a range of male grooming products   
1   A learning platform that provides topic-based ...   
2             It aims to make learning fun via games.   
3               Provides interior designing solutions   
4   It is a seed company engaged in production, pr...   
..                                                ...   
84  It connects client requirements to their suppl...   
85                Provides rooms for comfortable stay   
86  Find automobile repair and maintenance service...   
87                                Online car retailer   
88              Organised Non-Banking Finance Company   

                                             Founders  \
0                                  Shantanu Deshpande   
1               Adamas Belva Syah Devara, Iman Usman.   
2                                       Jatin Solanki   
3                        Srikanth Iyer, Rama Harinath   
4                              Narayana Reddy Punyala   
..                                                ...   
84                    Aaditya Sharda, Souvik Sengupta   
85                                     Ritesh Agarwal   
86  Amit Bhasin, Kushal Karwa, Nitin Rana, Rishabh...   
87  Niraj Singh, Ramanshu Mahaur, Ganesh Pawar, Mo...   
88                                     Rajendra Setia   

                                             Investor     Amount($)  \
0                                Sixth Sense Ventures    $6,300,000   
1                                    General Atlantic  $150,000,000   
2      Deepak Parekh, Amitabh Bachchan, Piyush Pandey   $28,000,000   
3   Evolvence India Fund (EIF), Pidilite Group, FJ...   $30,000,000   
4            Innovation in Food and Agriculture (IFA)    $6,000,000   
..                                                ...           ...   
84  Tiger Global, Nexus Venture Partners, Accel Pa...   $20,000,000   
85  MyPreferred Transformation, Avendus Finance, S...  $693,000,000   
86                                    Sequoia Capital    $5,000,000   
87  Norwest Venture Partners, General Catalyst, Fu...   $50,000,000   
88     TPG, Norwest Venture Partners, Evolvence India   $33,000,000   

            Stage  
0             NaN  
1        Series C  
2   Fresh funding  
3        Series D  
4             NaN  
..            ...  
84       Series A  
85            NaN  
86       Series B  
87            NaN  
88            NaN  

[89 rows x 9 columns]

## Accessing a csv file containing 2018 data from the root directory of this project

In [12]:
data_2018 = pd.read_csv("startup_funding2018.csv")

data_2018

Company Name                                           Industry  \
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

    Round/Series       Amount                          Location  \
0           Seed       250000       Bangalore, Karnataka, India   
1           Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2       Series A  ₹65,000,000           Gurgaon, Haryana, India   
3          Angel      2000000       Noida, Uttar Pradesh, India   
4           Seed            —  Hyderabad, Andhra Pradesh, India   
..           ...          ...                               ...   
521     Series C    225000000       Bangalore, Karnataka, India   
522     Series A            —           Haryana, Haryana, India   
523         Seed         7500        Mumbai, Maharashtra, India   
524         Seed  ₹35,000,000        Mumbai, Maharashtra, India   
525     Series C     35000000        Chennai, Tamil Nadu, India   

                                         About Company  
0    TheCollegeFever is a hub for fun, fiesta and f...  
1    A startup which aggregates milk from dairy far...  
2            Leading Online Loans Marketplace in India  
3    PayMe India is an innovative FinTech organizat...  
4    Eunimart is a one stop solution for merchants ...  
..                                                 ...  
521  Udaan is a B2B trade platform, designed specif...  
522            HappyEasyGo is an online travel domain.  
523  Mombay is a unique opportunity for housewives ...  
524  Droni Tech manufacture UAVs and develop softwa...  
525       Welcome to India's most convenient pharmacy!  

[526 rows x 6 columns]

# **Data preparation** 

## Inspecting all four datasets

In [13]:
data_2021.shape

(1209, 9)

In [14]:
data_2020.shape

(1055, 10)

In [15]:
data_2019.shape

(89, 9)

In [16]:
data_2018.shape

(526, 6)

## Checking the datatype the number of columns of the four datasets using the **.info()** method

In [17]:
data_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1209 entries, 0 to 1208
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1209 non-null   object 
 1   Founded        1208 non-null   float64
 2   HeadQuarter    1208 non-null   object 
 3   Sector         1209 non-null   object 
 4   What_it_does   1209 non-null   object 
 5   Founders       1205 non-null   object 
 6   Investor       1147 non-null   object 
 7   Amount         1206 non-null   object 
 8   Stage          781 non-null    object 
dtypes: float64(1), object(8)
memory usage: 85.1+ KB


In [18]:
data_2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1055 entries, 0 to 1054
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1055 non-null   object 
 1   Founded        842 non-null    float64
 2   HeadQuarter    961 non-null    object 
 3   Sector         1042 non-null   object 
 4   What_it_does   1055 non-null   object 
 5   Founders       1043 non-null   object 
 6   Investor       1017 non-null   object 
 7   Amount         801 non-null    float64
 8   Stage          591 non-null    object 
 9   column10       2 non-null      object 
dtypes: float64(2), object(8)
memory usage: 82.5+ KB


In [19]:
data_2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89 entries, 0 to 88
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company/Brand  89 non-null     object 
 1   Founded        60 non-null     float64
 2   HeadQuarter    70 non-null     object 
 3   Sector         84 non-null     object 
 4   What it does   89 non-null     object 
 5   Founders       86 non-null     object 
 6   Investor       89 non-null     object 
 7   Amount($)      89 non-null     object 
 8   Stage          43 non-null     object 
dtypes: float64(1), object(8)
memory usage: 6.4+ KB


In [20]:
data_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company Name   526 non-null    object
 1   Industry       526 non-null    object
 2   Round/Series   526 non-null    object
 3   Amount         526 non-null    object
 4   Location       526 non-null    object
 5   About Company  526 non-null    object
dtypes: object(6)
memory usage: 24.8+ KB


## Checking for null values using the **.isnull().sum()**

In [21]:
data_2021.isnull().sum()

Company_Brand      0
Founded            1
HeadQuarter        1
Sector             0
What_it_does       0
Founders           4
Investor          62
Amount             3
Stage            428
dtype: int64

In [22]:
data_2020.isnull().sum()

Company_Brand       0
Founded           213
HeadQuarter        94
Sector             13
What_it_does        0
Founders           12
Investor           38
Amount            254
Stage             464
column10         1053
dtype: int64

In [23]:
data_2019.isnull().sum()

Company/Brand     0
Founded          29
HeadQuarter      19
Sector            5
What it does      0
Founders          3
Investor          0
Amount($)         0
Stage            46
dtype: int64

In [24]:
data_2018.isnull().sum()

Company Name     0
Industry         0
Round/Series     0
Amount           0
Location         0
About Company    0
dtype: int64

## After careful analysis **"column10"** is dropped because it contains no values

In [25]:
data_2020_1 = data_2020.drop(["column10"],axis=1,inplace=True)

In [26]:
data_2020.shape

(1055, 9)

## Adding an additional column called **"Funding Year"** to show the year the companny got funds

In [27]:
data_2021["Funding_Year"] = 2021

data_2021.head(15)

Company_Brand  Founded HeadQuarter            Sector  \
0   Unbox Robotics   2019.0   Bangalore        AI startup   
1           upGrad   2015.0      Mumbai            EdTech   
2      Lead School   2012.0      Mumbai            EdTech   
3          Bizongo   2015.0      Mumbai    B2B E-commerce   
4         FypMoney   2021.0    Gurugram           FinTech   
5    Urban Company   2014.0   New Delhi     Home services   
6   Comofi Medtech   2018.0   Bangalore        HealthTech   
7      Qube Health   2016.0      Mumbai        HealthTech   
8         Vitra.ai   2020.0   Bangalore      Tech Startup   
9           Taikee   2010.0      Mumbai        E-commerce   
10       Fitterfly   2016.0      Mumbai        HealthTech   
11       FreeStand   2017.0   New Delhi       B2B service   
12    MyDentalPlan   2014.0   Bangalore        Helathcare   
13    Freyr Energy   2014.0   Hyderabad  Renewable Energy   
14       DealShare   2018.0      Jaipur        E-commerce   

                                         What_it_does  \
0   Unbox Robotics builds on-demand AI-driven ware...   
1      UpGrad is an online higher education platform.   
2   LEAD School offers technology based school tra...   
3   Bizongo is a business-to-business online marke...   
4   FypMoney is Digital NEO Bank for Teenagers, em...   
5   Urban Company (Formerly UrbanClap) is a home a...   
6    Comofi Medtech is a healthcare robotics startup.   
7   India's Most Respected Workplace Healthcare Ma...   
8   Vitra.ai is an AI-based video translation plat...   
9   Taikee is the ISO-certified, B2B e-commerce pl...   
10  Fitterfly offers customized and personalized w...   
11  FreeStand enables FMCG brands to execute track...   
12  My Dental Plan provides dental care packages a...   
13  Freyr Energy is a company that provides full s...   
14             DealShare is a Social Commerce Startup   

                                             Founders  \
0                         Pramod Ghadge, Shahid Memon   
1   Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                          Smita Deorah, Sumeet Mehta   
3             Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                       Kapil Banwari   
5   Abhiraj Singh Bhal, Raghav Chandra, Varun Khaitan   
6                                          Gururaj KB   
7                                         Gagan Kapur   
8                                      Akash Nidhi PS   
9                  Nidhi Ramachandran, Sachin Chhabra   
10                                 Dr Arbinder Singal   
11                           Konark Sharma, Sneh Soni   
12  Dr Mohender Narula, Dr AnandKrishna, Dr Girish...   
13                    Radhika Choudary, Saurabh Marda   
14          Sankar Bora, Sourjyendu Medda, Vineet Rao   

                                            Investor        Amount  \
0                        BEENEXT, Entrepreneur First    $1,200,000   
1           Unilazer Ventures, IIFL Asset Management  $120,000,000   
2                   GSV Ventures, Westbridge Capital   $30,000,000   
3                             CDC Group, IDG Capital   $51,000,000   
4      Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000   
5                                         Vy Capital  $188,000,000   
6                                  CIIE.CO, KIIT-TBI      $200,000   
7                          Inflection Point Ventures   Undisclosed   
8                                  Inflexor Ventures   Undisclosed   
9                                               None    $1,000,000   
10     9Unicorns Accelerator Fund, Metaform Ventures    $3,000,000   
11  SucSEED Indovation, IIM Calcutta Innovation Park      $100,000   
12                              Safe Planet Medicare      $700,000   
13                     Impact Partners, C4D Partners    $2,000,000   
14          Tiger Global Management, InnoVen Capital    $9,000,000   

           Stage  Funding_Year  
0   Pre-series A          2021  
1           Non

In [28]:
data_2020["Funding_Year"] = 2020

data_2020.head(15)

Company_Brand  Founded HeadQuarter               Sector  \
0     Aqgromalin   2019.0     Chennai             AgriTech   
1       Krayonnz   2019.0   Bangalore               EdTech   
2   PadCare Labs   2018.0        Pune   Hygiene management   
3          NCOME   2020.0   New Delhi               Escrow   
4     Gramophone   2016.0      Indore             AgriTech   
5         qZense   2019.0   Bangalore             AgriTech   
6   MyClassboard   2008.0   Hyderabad               EdTech   
7          Metvy   2018.0     Gurgaon  Networking platform   
8         Rupeek   2015.0   Bangalore              FinTech   
9      Gig India   2017.0        Pune        Crowdsourcing   
10   Slurrp Farm   2016.0     Gurgaon     Food & Bevarages   
11        Medfin   2017.0   Bangalore           HealthTech   
12       MasterG   2015.0   New Delhi      Fashion startup   
13      Brila 91   2014.0   New Delhi     Food & Bevarages   
14    FoodyBuddy   2015.0     Belgaum        Food Industry   

                                         What_it_does  \
0                        Cultivating Ideas for Profit   
1   An academy-guardian-scholar centric ecosystem ...   
2    Converting bio-hazardous waste to harmless waste   
3                        Escrow-as-a-service platform   
4   Gramophone is an AgTech platform enabling acce...   
5   qZense Labs is building the next-generation Io...   
6   MyClassboard is a full-fledged School / Colleg...   
7   AI driven networking platform for individuals ...   
8   Rupeek is an online lending platform that spec...   
9   GigIndia is a marketplace that provides on-dem...   
10       Slurrp Farm is a children health food brand.   
11  Medfin is a Discovery platform for medical pro...   
12  MasterG is a design and skill development orga...   
13                   Bira 91 is a craft beer company.   
14        FoodyBuddy is a online food delivery portal   

                                             Founders  \
0                     Prasanna Manogaran, Bharani C L   
1                    Saurabh Dixit, Gurudutt Upadhyay   
2                                     Ajinkya Dhariya   
3                                       Ritesh Tiwari   
4   Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   
5                        Rubal Chib, Dr Srishti Batra   
6                                      Ajay Sakhamuri   
7                                    Shawrya Mehrotra   
8             Amar Prabhu, Ashwin Soni, Sumit Maniyar   
9                        Aditya Shirole, Sahil Sharma   
10                                    Meghana Narayan   
11           Arjun Kumar, Arun Kumar, Sidharth Gurjar   
12                                      Gayatri Jolly   
13                                         Ankur Jain   
14         Akil Sethuraman, Anup Gopinath, Rachna Rao   

                                         Investor      Amount         Stage  \
0                                 Angel investors    200000.0          None   
1                                 GSF Accelerator    100000.0      Pre-seed   
2                                  Venture Center         NaN      Pre-seed   
3             Venture Catalysts, PointOne Capital    400000.0          None   
4             Siana Capital Management, Info Edge    340000.0          None   
5   Venture Catalysts, 9Unicorns Accelerator Fund    600000.0          Seed   
6                                     ICICI Bank.    600000.0  Pre-series A   
7                                      HostelFund         NaN    Pre-series   
8    KB Investment, Bertelsmann India Investments  45000000.0      Series C   
9       Shantanu Deshpande, Subramaniam Ramadorai   1000000.0  Pre-series A   
10                              Fireside Ventures   2000000.0      Series A   
11                 Axilor Ventures, S.D. Shibulal         NaN      Series A   
12                  Acumen Fund's angel programme         NaN          None   
13    Sequoia Capital India, Sixth Sense Ventures   1200000.0          None   

In [29]:
data_2019["Funding_Year"] = 2019

data_2019.head(15)

Company/Brand  Founded HeadQuarter           Sector  \
0   Bombay Shaving      NaN         NaN        Ecommerce   
1        Ruangguru   2014.0      Mumbai           Edtech   
2         Eduisfun      NaN      Mumbai           Edtech   
3         HomeLane   2014.0     Chennai  Interior design   
4         Nu Genes   2004.0   Telangana         AgriTech   
5         FlytBase      NaN        Pune       Technology   
6            Finly      NaN   Bangalore             SaaS   
7         Kratikal   2013.0       Noida       Technology   
8        Quantiphi      NaN         NaN        AI & Tech   
9         Lenskart   2010.0       Delhi       E-commerce   
10      Cub McPaws   2010.0      Mumbai  E-commerce & AR   
11        Appnomic      NaN   Bangalore             SaaS   
12      Kaleidofin   2018.0     Chennai          Fintech   
13       JobSquare   2019.0   Ahmedabad          HR tech   
14     Open Secret      NaN         NaN        Food tech   

                                         What it does  \
0          Provides a range of male grooming products   
1   A learning platform that provides topic-based ...   
2             It aims to make learning fun via games.   
3               Provides interior designing solutions   
4   It is a seed company engaged in production, pr...   
5                         A drone automation platform   
6   It builds software products that makes work si...   
7   It is a product-based cybersecurity solutions ...   
8   It is an AI and big data services company prov...   
9                             It is a eyewear company   
10  A B2C  brand that focusses on premium and comf...   
11  It is a self-healing enterprise, the IT operat...   
12  Platform that provides complete financial solu...   
13  Technology-based platform that is connecting s...   
14           It produces and sells top quality snacks   

                                           Founders  \
0                                Shantanu Deshpande   
1             Adamas Belva Syah Devara, Iman Usman.   
2                                     Jatin Solanki   
3                      Srikanth Iyer, Rama Harinath   
4                            Narayana Reddy Punyala   
5                                       Nitin Gupta   
6                         Vivek AG, Veekshith C Rai   
7   Pavan Kushwaha, Paratosh Bansal, Dip Jung Thapa   
8                                    Renuka Ramnath   
9      Peyush Bansal, Amit Chaudhary, Sumeet Kapahi   
10                          Abhay Bhat, Kinnar Shah   
11                                    D Padmanabhan   
12                Puneet Gupta, Sucharita Mukherjee   
13                                     Ishit Jethwa   
14                      Ahana Gautam, Udit Kejriwal   

                                             Investor     Amount($)  \
0                                Sixth Sense Ventures    $6,300,000   
1                                    General Atlantic  $150,000,000   
2      Deepak Parekh, Amitabh Bachchan, Piyush Pandey   $28,000,000   
3   Evolvence India Fund (EIF), Pidilite Group, FJ...   $30,000,000   
4            Innovation in Food and Agriculture (IFA)    $6,000,000   
5                                         Undisclosed   Undisclosed   
6   Social Capital, AngelList India, Gemba Capital...   Undisclosed   
7          Gilda VC, Art Venture, Rajeev Chitrabhanu.    $1,000,000   
8                Multiples Alternate Asset Management   $20,000,000   
9                                            SoftBank  $275,000,000   
10                                  Venture Catalysts   Undisclosed   
11                                   Avataar Ventures   $22,000,000   
12                                         Oikocredit    $5,000,000   
13                                      Titan Capital      $140,500   
14                                    Matrix Partners   Undisclosed   

            Stage  Funding_Year  
0             NaN          2019  
1        Series C          2019  
2   Fresh f

In [30]:
data_2018["Funding_Year"] = 2018

data_2018.head(15)

Company Name  \
0              TheCollegeFever   
1              Happy Cow Dairy   
2                   MyLoanCare   
3                  PayMe India   
4                     Eunimart   
5                       Hasura   
6                    Tripshelf   
7                 Hyperdata.IO   
8                 Freightwalla   
9           Microchip Payments   
10  BizCrum Infotech Pvt. Ltd.   
11                     Emojifi   
12                       Flock   
13                   Freshboxx   
14  Wide Mobility Mechatronics   

                                             Industry Round/Series  \
0   Brand Marketing, Event Promotion, Marketing, S...         Seed   
1                                Agriculture, Farming         Seed   
2    Credit, Financial Services, Lending, Marketplace     Series A   
3                         Financial Services, FinTech        Angel   
4                  E-Commerce Platforms, Retail, SaaS         Seed   
5                    Cloud Infrastructure, PaaS, SaaS         Seed   
6                      Internet, Leisure, Marketplace         Seed   
7                                     Market Research        Angel   
8        Information Services, Information Technology         Seed   
9                                     Mobile Payments         Seed   
10                                         B2B, Shoes         Seed   
11                                           Internet         Seed   
12  Apps, Collaboration, Developer Platform, Enter...         Seed   
13                                      Food Delivery         Seed   
14                              Industrial Automation         Seed   

         Amount                          Location  \
0        250000       Bangalore, Karnataka, India   
1   ₹40,000,000        Mumbai, Maharashtra, India   
2   ₹65,000,000           Gurgaon, Haryana, India   
3       2000000       Noida, Uttar Pradesh, India   
4             —  Hyderabad, Andhra Pradesh, India   
5       1600000       Bengaluru, Karnataka, India   
6   ₹16,000,000             Kalkaji, Delhi, India   
7   ₹50,000,000  Hyderabad, Andhra Pradesh, India   
8             —        Mumbai, Maharashtra, India   
9             —       Bangalore, Karnataka, India   
10            —               Delhi, Delhi, India   
11            —       Bengaluru, Karnataka, India   
12            —                       India, Asia   
13            —           Hubli, Karnataka, India   
14            —       Bangalore, Karnataka, India   

                                        About Company  Funding_Year  
0   TheCollegeFever is a hub for fun, fiesta and f...          2018  
1   A startup which aggregates milk from dairy far...          2018  
2           Leading Online Loans Marketplace in India          2018  
3   PayMe India is an innovative FinTech organizat...          2018  
4   Eunimart is a one stop solution for merchants ...          2018  
5   Hasura is a platform that allows developers to...          2018  
6   Tripshelf is an online market place for holida...          2018  
7   Hyperdata combines advanced machine learning w...          2018  
8   Freightwalla is an international forwarder tha...          2018  
9   Microchip payments is a mobile-based payment a...          2018  
10           Building Transactionary B2B Marketplaces          2018  
11  Emojifi is an app that provides live emoji, st...          2018  
12  Flock is a real-time messaging and collaborati...          2018  
13  Freshboxx is incubated at Sandbox Startups, a ...          2018  
14  Wide Mobility Mechatronics is a Private Compan...          2018

## Renaming columns to facilitate merging 

In [31]:
data_2021.rename(columns={"Company_Brand" : "CompanyName"}, inplace = True)

data_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1209 entries, 0 to 1208
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   CompanyName   1209 non-null   object 
 1   Founded       1208 non-null   float64
 2   HeadQuarter   1208 non-null   object 
 3   Sector        1209 non-null   object 
 4   What_it_does  1209 non-null   object 
 5   Founders      1205 non-null   object 
 6   Investor      1147 non-null   object 
 7   Amount        1206 non-null   object 
 8   Stage         781 non-null    object 
 9   Funding_Year  1209 non-null   int64  
dtypes: float64(1), int64(1), object(8)
memory usage: 94.6+ KB


In [32]:
data_2020.rename(columns={"Company_Brand" : "CompanyName"}, inplace = True)

data_2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1055 entries, 0 to 1054
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   CompanyName   1055 non-null   object 
 1   Founded       842 non-null    float64
 2   HeadQuarter   961 non-null    object 
 3   Sector        1042 non-null   object 
 4   What_it_does  1055 non-null   object 
 5   Founders      1043 non-null   object 
 6   Investor      1017 non-null   object 
 7   Amount        801 non-null    float64
 8   Stage         591 non-null    object 
 9   Funding_Year  1055 non-null   int64  
dtypes: float64(2), int64(1), object(7)
memory usage: 82.5+ KB


In [33]:
data_2019.rename(columns={"Company/Brand" : "CompanyName"}, inplace = True)

data_2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89 entries, 0 to 88
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   CompanyName   89 non-null     object 
 1   Founded       60 non-null     float64
 2   HeadQuarter   70 non-null     object 
 3   Sector        84 non-null     object 
 4   What it does  89 non-null     object 
 5   Founders      86 non-null     object 
 6   Investor      89 non-null     object 
 7   Amount($)     89 non-null     object 
 8   Stage         43 non-null     object 
 9   Funding_Year  89 non-null     int64  
dtypes: float64(1), int64(1), object(8)
memory usage: 7.1+ KB


In [34]:
data_2018.rename(columns={"Company Name" : "CompanyName"}, inplace=True)

data_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   CompanyName    526 non-null    object
 1   Industry       526 non-null    object
 2   Round/Series   526 non-null    object
 3   Amount         526 non-null    object
 4   Location       526 non-null    object
 5   About Company  526 non-null    object
 6   Funding_Year   526 non-null    int64 
dtypes: int64(1), object(6)
memory usage: 28.9+ KB


In [35]:
## close the database connection 

connection.close()

In [36]:
data_2021.rename(columns={"Amount" : "Fund_Amount"}, inplace=True)

data_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1209 entries, 0 to 1208
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   CompanyName   1209 non-null   object 
 1   Founded       1208 non-null   float64
 2   HeadQuarter   1208 non-null   object 
 3   Sector        1209 non-null   object 
 4   What_it_does  1209 non-null   object 
 5   Founders      1205 non-null   object 
 6   Investor      1147 non-null   object 
 7   Fund_Amount   1206 non-null   object 
 8   Stage         781 non-null    object 
 9   Funding_Year  1209 non-null   int64  
dtypes: float64(1), int64(1), object(8)
memory usage: 94.6+ KB


In [37]:
data_2020.rename(columns={"Amount" : "Fund_Amount"}, inplace=True)

data_2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1055 entries, 0 to 1054
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   CompanyName   1055 non-null   object 
 1   Founded       842 non-null    float64
 2   HeadQuarter   961 non-null    object 
 3   Sector        1042 non-null   object 
 4   What_it_does  1055 non-null   object 
 5   Founders      1043 non-null   object 
 6   Investor      1017 non-null   object 
 7   Fund_Amount   801 non-null    float64
 8   Stage         591 non-null    object 
 9   Funding_Year  1055 non-null   int64  
dtypes: float64(2), int64(1), object(7)
memory usage: 82.5+ KB


In [38]:
data_2019.rename(columns={"Amount($)" : "Fund_Amount"}, inplace=True)

data_2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89 entries, 0 to 88
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   CompanyName   89 non-null     object 
 1   Founded       60 non-null     float64
 2   HeadQuarter   70 non-null     object 
 3   Sector        84 non-null     object 
 4   What it does  89 non-null     object 
 5   Founders      86 non-null     object 
 6   Investor      89 non-null     object 
 7   Fund_Amount   89 non-null     object 
 8   Stage         43 non-null     object 
 9   Funding_Year  89 non-null     int64  
dtypes: float64(1), int64(1), object(8)
memory usage: 7.1+ KB


In [39]:
data_2018.rename(columns={"Amount" : "Fund_Amount"}, inplace=True)

data_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   CompanyName    526 non-null    object
 1   Industry       526 non-null    object
 2   Round/Series   526 non-null    object
 3   Fund_Amount    526 non-null    object
 4   Location       526 non-null    object
 5   About Company  526 non-null    object
 6   Funding_Year   526 non-null    int64 
dtypes: int64(1), object(6)
memory usage: 28.9+ KB


In [40]:
data_2018.rename(columns={"Industry" : "Sector"}, inplace=True)

data_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   CompanyName    526 non-null    object
 1   Sector         526 non-null    object
 2   Round/Series   526 non-null    object
 3   Fund_Amount    526 non-null    object
 4   Location       526 non-null    object
 5   About Company  526 non-null    object
 6   Funding_Year   526 non-null    int64 
dtypes: int64(1), object(6)
memory usage: 28.9+ KB


In [41]:
data_2018.rename(columns={"Round/Series" : "Stage"}, inplace=True)

data_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   CompanyName    526 non-null    object
 1   Sector         526 non-null    object
 2   Stage          526 non-null    object
 3   Fund_Amount    526 non-null    object
 4   Location       526 non-null    object
 5   About Company  526 non-null    object
 6   Funding_Year   526 non-null    int64 
dtypes: int64(1), object(6)
memory usage: 28.9+ KB


In [42]:
data_2018.rename(columns={"Location" : "HeadQuarter"}, inplace=True)

data_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   CompanyName    526 non-null    object
 1   Sector         526 non-null    object
 2   Stage          526 non-null    object
 3   Fund_Amount    526 non-null    object
 4   HeadQuarter    526 non-null    object
 5   About Company  526 non-null    object
 6   Funding_Year   526 non-null    int64 
dtypes: int64(1), object(6)
memory usage: 28.9+ KB


In [43]:
data_2019.rename(columns={"What it does" : "What_it_does"}, inplace=True)

data_2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89 entries, 0 to 88
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   CompanyName   89 non-null     object 
 1   Founded       60 non-null     float64
 2   HeadQuarter   70 non-null     object 
 3   Sector        84 non-null     object 
 4   What_it_does  89 non-null     object 
 5   Founders      86 non-null     object 
 6   Investor      89 non-null     object 
 7   Fund_Amount   89 non-null     object 
 8   Stage         43 non-null     object 
 9   Funding_Year  89 non-null     int64  
dtypes: float64(1), int64(1), object(8)
memory usage: 7.1+ KB


In [44]:
data_2018.rename(columns={"About Company" : "What_it_does"}, inplace=True)

data_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   CompanyName   526 non-null    object
 1   Sector        526 non-null    object
 2   Stage         526 non-null    object
 3   Fund_Amount   526 non-null    object
 4   HeadQuarter   526 non-null    object
 5   What_it_does  526 non-null    object
 6   Funding_Year  526 non-null    int64 
dtypes: int64(1), object(6)
memory usage: 28.9+ KB


## Droping Columns to facilitate Merging 

In [45]:
columns_to_drop = ["Founded", "Founders", "Investor"]

data_2019 = data_2019.drop(columns=columns_to_drop)

data_2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89 entries, 0 to 88
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   CompanyName   89 non-null     object
 1   HeadQuarter   70 non-null     object
 2   Sector        84 non-null     object
 3   What_it_does  89 non-null     object
 4   Fund_Amount   89 non-null     object
 5   Stage         43 non-null     object
 6   Funding_Year  89 non-null     int64 
dtypes: int64(1), object(6)
memory usage: 5.0+ KB


In [46]:
columns_to_drop = ["Founded", "Founders", "Investor"]

data_2020 = data_2020.drop(columns=columns_to_drop)

data_2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1055 entries, 0 to 1054
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   CompanyName   1055 non-null   object 
 1   HeadQuarter   961 non-null    object 
 2   Sector        1042 non-null   object 
 3   What_it_does  1055 non-null   object 
 4   Fund_Amount   801 non-null    float64
 5   Stage         591 non-null    object 
 6   Funding_Year  1055 non-null   int64  
dtypes: float64(1), int64(1), object(5)
memory usage: 57.8+ KB


In [47]:
columns_to_drop = ["Founded", "Founders", "Investor"]

data_2021 = data_2021.drop(columns=columns_to_drop)

data_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1209 entries, 0 to 1208
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   CompanyName   1209 non-null   object
 1   HeadQuarter   1208 non-null   object
 2   Sector        1209 non-null   object
 3   What_it_does  1209 non-null   object
 4   Fund_Amount   1206 non-null   object
 5   Stage         781 non-null    object
 6   Funding_Year  1209 non-null   int64 
dtypes: int64(1), object(6)
memory usage: 66.2+ KB


## Merge all datasets to one

In [48]:
joined_data = pd.concat([data_2018, data_2019, data_2020, data_2021], ignore_index=True)

joined_data.info

<bound method DataFrame.info of           CompanyName                                             Sector  \
0     TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1     Happy Cow Dairy                               Agriculture, Farming   
2          MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3         PayMe India                        Financial Services, FinTech   
4            Eunimart                 E-Commerce Platforms, Retail, SaaS   
...               ...                                                ...   
2874         Gigforce                              Staffing & Recruiting   
2875           Vahdam                                   Food & Beverages   
2876     Leap Finance                                 Financial Services   
2877     CollegeDekho                                             EdTech   
2878           WeRize                                 Financial Services   

             Stage  Fund_Amount                       H

In [49]:
joined_data.to_csv(r"C:\Users\Safowaa\Documents\Azibiafrica\AzubiPython\Indian_start-up_ecosystem\joined_data.csv", index=False)


In [121]:
combined21_18 = pd.read_csv("joined_data.csv")

combined21_18.head(20)

CompanyName  \
0                    TheCollegeFever   
1                    Happy Cow Dairy   
2                         MyLoanCare   
3                        PayMe India   
4                           Eunimart   
5                             Hasura   
6                          Tripshelf   
7                       Hyperdata.IO   
8                       Freightwalla   
9                 Microchip Payments   
10        BizCrum Infotech Pvt. Ltd.   
11                           Emojifi   
12                             Flock   
13                         Freshboxx   
14        Wide Mobility Mechatronics   
15                           Pitstop   
16                            Mihuru   
17                              Fyle   
18                          AppWharf   
19  Antariksh Waste Ventures Pvt ltd   

                                               Sector     Stage   Fund_Amount  \
0   Brand Marketing, Event Promotion, Marketing, S...      Seed        250000   
1                                Agriculture, Farming      Seed   ₹40,000,000   
2    Credit, Financial Services, Lending, Marketplace  Series A   ₹65,000,000   
3                         Financial Services, FinTech     Angel       2000000   
4                  E-Commerce Platforms, Retail, SaaS      Seed             —   
5                    Cloud Infrastructure, PaaS, SaaS      Seed       1600000   
6                      Internet, Leisure, Marketplace      Seed   ₹16,000,000   
7                                     Market Research     Angel   ₹50,000,000   
8        Information Services, Information Technology      Seed             —   
9                                     Mobile Payments      Seed             —   
10                                         B2B, Shoes      Seed             —   
11                                           Internet      Seed             —   
12  Apps, Collaboration, Developer Platform, Enter...      Seed             —   
13                                      Food Delivery      Seed             —   
14                              Industrial Automation      Seed             —   
15        Automotive, Search Engine, Service Industry      Seed  ₹100,000,000   
16                          Finance, Internet, Travel      Seed        150000   
17  Accounting, Business Information Systems, Busi...      Seed       1100000   
18  Artificial Intelligence, Product Search, SaaS,...      Seed             —   
19               Internet of Things, Waste Management      Seed      ₹500,000   

                         HeadQuarter  \
0        Bangalore, Karnataka, India   
1         Mumbai, Maharashtra, India   
2            Gurgaon, Haryana, India   
3        Noida, Uttar Pradesh, India   
4   Hyderabad, Andhra Pradesh, India   
5        Bengaluru, Karnataka, India   
6              Kalkaji, Delhi, India   
7   Hyderabad, Andhra Pradesh, India   
8         Mumbai, Maharashtra, India   
9        Bangalore, Karnataka, India   
10               Delhi, Delhi, India   
11       Bengaluru, Karnataka, India   
12                       India, Asia   
13           Hubli, Karnataka, India   
14       Bangalore, Karnataka, India   
15       Bengaluru, Karnataka, India   
16        Mumbai, Maharashtra, India   
17       Bengaluru, Karnataka, India   
18           New Delhi, Delhi, India   
19        Chennai, Tamil Nadu, India   

                                         What_it_does  Funding_Year  
0   TheCollegeFever is a hub for fun, fiesta and f...          2018  
1   A startup which aggregates milk from dairy far...          2018  
2           Leading Online Loans Marketplace in India          2018  
3   PayMe India is an innovative FinTech organizat...          2018  
4   Eunimart is a one stop solution for merchants ...          2018  
5   Hasura is a platform that allows developers to...          2018  
6   Tripshelf is an online market place for holida...          2018  
7   Hyperdata combines advanced machine learning w...          2018  
8   Freight

In [122]:
combined21_18.sample(n=20)

CompanyName                                        Sector  \
916                   Equiwatt                                  Tech company   
69                     Enakshi                                       Fashion   
1430                    SaScan                                    Healthcare   
1254             LetsTransport                                      Logitech   
70    Ultraviolette Automotive  Automotive, Electric Vehicle, Energy Storage   
1417             RaRa Delivery                                     Logistics   
935      Mobile Premier League                                 Entertainment   
281        Inntot Technologies                          Consumer Electronics   
742                    SignalX                                    AI startup   
106            Vivriti Capital                            Financial Services   
2332                  Bimaplan                                    Insuretech   
554                  HungerBox                                      Foodtech   
763                  Oga Fit's                               Fitness startup   
211                    Chezuba             Internet, Social Entrepreneurship   
1274                    QuaQua                                        Travel   
2391                     Voiro                                 Advertisement   
2174                     Otipy                              Food & Beverages   
2541                 TurboHire                             Computer Software   
2452                 Riskcovry                                    InsureTech   
2140                   Murf.ai                                  SaaS startup   

                         Stage   Fund_Amount  \
916                        NaN      300000.0   
69                        Seed    ₹8,000,000   
1430                       NaN           NaN   
1254                       NaN     1313500.0   
70                    Series A       ₹60,000   
1417                Seed Round      800000.0   
935                        NaN    74000000.0   
281                       Seed             —   
742                       Seed      800000.0   
106   Venture - Series Unknown   $28,500,000   
2332              Pre-series A      $2500000   
554                   Series C   $12,000,000   
763               Pre-series A           NaN   
211                      Angel    ₹1,400,000   
1274                       NaN     1000000.0   
2391              Pre-series A    $1,800,000   
2174                  Series A     $10200000   
2541                       NaN  $Undisclosed   
2452                  Series A    $5,000,000   
2140                      Seed      $1500000   

                                            HeadQuarter  \
916   Newcastle Upon Tyne, Newcastle upon Tyne, Unit...   
69                            Ahmedabad, Gujarat, India   
1430                                          Bangalore   
1254                                          Bangalore   
70                          Bangalore, Karnataka, India   
1417                                                NaN   
935                                           Bangalore   
281                                Kochi, Kerala, India   
742                                           Hyderabad   
106                          Chennai, Tamil Nadu, India   
2332                                          Bangalore   
554                                           Bangalore   
763                                           Bangalore   
211                    Hyderabad, Andhra Pradesh, India   
1274                                          Hyderabad   
2391                                          Bangalore   
2174                                           Gurugram   
2541                                          Hyderabad   
2452                                          Bangalore   
2140                                          Bangalore   

                                           What_it_does  Funding_Year  
916   Equiwatt is

## Checking for columns with empty and / null rows

In [123]:
combined21_18.isna().any()

CompanyName     False
Sector           True
Stage            True
Fund_Amount      True
HeadQuarter      True
What_it_does    False
Funding_Year    False
dtype: bool

In [124]:
combined21_18.isnull().any()

CompanyName     False
Sector           True
Stage            True
Fund_Amount      True
HeadQuarter      True
What_it_does    False
Funding_Year    False
dtype: bool

## Checking for the number of empty rows in each column

In [125]:
if combined21_18["Sector"].isna().any():
    print(combined21_18["Sector"].isna().sum())
else:
    print("There are no empty rows in the column.")

18


In [126]:
if combined21_18["Stage"].isna().any():
    print(combined21_18["Stage"].isna().sum())
else:
    print("There are no empty rows in the column.")

938


In [127]:
if combined21_18["Fund_Amount"].isna().any():
    print(combined21_18["Fund_Amount"].isna().sum())
else:
    print("There are no empty rows in the column.")


257


In [128]:
if combined21_18["HeadQuarter"].isna().any():
    print(combined21_18["HeadQuarter"].isna().sum())
else:
    print("There are no empty rows in the column.")

114


In [129]:
# Calculate the percentage of missing rows
total_rows = len(combined21_18)
missing_rows = combined21_18.isnull().any(axis=1).sum()
missing_percentage = (missing_rows / total_rows) * 100

print(f"Percentage of missing rows: {missing_percentage:.2f}%")

Percentage of missing rows: 38.59%


### Percentage of missing rows: 38.59%

## Since the percentage of missing data is more than 10% we will fill them with the mode/median of each column

## From our findings above the "Sector" column has 18 empty rows so  we fill it with the most common category

In [130]:
# First we take a look at a sample
combined21_18["Sector"].sample(n=15)

411                         Fashion, Jewelry
513                Beauty, Fashion, Wellness
796                              QSR startup
1251                         Yoga & wellness
38      E-Commerce, Fashion, Jewelry, Retail
1952                   Industrial Automation
1083                              E-commerce
1037                                  EdTech
908                                   Gaming
939                                  FinTech
1999                                 FinTech
99            Health Care, Hospital, Medical
1746                                 FinTech
2874                   Staffing & Recruiting
1831                      Financial Services
Name: Sector, dtype: object

In [131]:
# Convert all values in the "Sector" column to strings and handle NaN's and empty strings

combined21_18["Sector"] = combined21_18["Sector"].fillna("Unknown").replace({"": "Unknown", "—": "Unknown"})

In [132]:
# Split the strings into lists
combined21_18["Sector"] = combined21_18["Sector"].str.split(", ")

In [133]:
# Flatten the lists into a single list
split_sector = combined21_18["Sector"].explode()

In [134]:
# Determine the most common category
most_common_sector = split_sector.value_counts().idxmax()

In [135]:
# Fill missing or empty rows with the most common category
combined21_18["Sector"].fillna(most_common_sector, inplace=True)
combined21_18["Sector"] = combined21_18["Sector"].replace("", most_common_sector)
combined21_18["Sector"] = combined21_18["Sector"].apply(lambda x: most_common_sector if isinstance(x, str) and ',' not in x else x)

In [151]:
# Test to see if there are any missing values
combined21_18["Sector"].isna().any()
combined21_18["Sector"].isnull().sum()

False

In [153]:
# Run samples to check the data
combined21_18["Sector"].sample(n=25)

965                                     [Entertainment]
1432                               [Health and Fitness]
2263                                        [Logistics]
2446                                      [Hospitality]
2579                 [Professional Training & Coaching]
1043                                             [Tech]
2421                                      [Health care]
1046                                      [Home Design]
863                                      [Tech Startup]
2789                                       [E-learning]
1511                                         [Foodtech]
1097                             [Linguistic Spiritual]
2699                                          [Finance]
1225                                          [Fintech]
528                                            [Edtech]
68      [Food and Beverage, Food Processing, Nutrition]
326                                           [Unknown]
247                                           [U

## From our findings earlier the "Stage" column has 938 empty rows so  we fill it with "Unknown"

In [155]:
combined21_18["Stage"] = combined21_18["Stage"].fillna("Unknown").replace("", "Unknown")

In [156]:
combined21_18["Stage"].isna().any()

False

In [157]:
# Run samples to check the data
combined21_18["Stage"].sample(n=25)

1067        Series C
43              Seed
252             Seed
698             Seed
1997            Seed
2742    Pre-series A
1355         Unknown
1375         Unknown
1017        Series C
1123    Pre Series A
2584            Seed
2492            Seed
723          Unknown
2115         Unknown
1241         Unknown
69              Seed
2832            Seed
1185         Unknown
521         Series C
2761         Unknown
2636            Seed
2853        Series A
2782        Pre-seed
1527         Unknown
691             Seed
Name: Stage, dtype: object

## From our findings earlier the "Fund_Amount" column has 257 empty rows so  we fill it with  its Median

#### But first we change the currency to our assumed currency which is dollar and remove the currency symbols. In our conversion we make use of the average exchange rate of each year. **source: [Google](https://www.exchangerates.org.uk/INR-USD-spot-exchange-rates-history.html)**

In [169]:
# Run samples to check the data
combined21_18["Fund_Amount"].sample(n=25)

91      ₹1,000,000,000
2714      $Undisclosed
1527               NaN
1033          650000.0
1046         9000000.0
378         $3,300,000
2628          $5000000
2465        $1,000,000
1342               NaN
287                  —
743           300000.0
1013        30000000.0
1734       Undisclosed
242       ₹103,000,000
101     $1,000,000,000
542       $540,000,000
597         $1,500,000
1266               NaN
2196          $5000000
794          1000000.0
1677       Undisclosed
1838         $28000000
2875         $20000000
1053          500000.0
31             1000000
Name: Fund_Amount, dtype: object

#### Replace Non-numeric Values with NaN

In [174]:
# Replace 'Undisclosed', '$Undisclosed', '—', '-', and other non-numeric values with NaN

combined21_18['Fund_Amount'] = combined21_18['Fund_Amount'].replace(['Undisclosed', '$Undisclosed', '—', '-'], np.nan)

#### Define Exchange Rates

In [176]:
# Exchange rates from INR to USD for each year
exchange_rates = {
    2018: 0.0146,
    2019: 0.0142,
    2020: 0.0135,
    2021: 0.0135
}


#### Define the Conversion Function

In [178]:
# Function to convert string amounts to float and handle currency symbols
def convert_currency(row):
    amount = row['Fund_Amount']
    year = row['Funding_Year']
    
    if pd.isna(amount):
        return np.nan
    amount = str(amount)
    
    if '₹' in amount:
        # Remove commas and ₹ symbol
        cleaned_amount = amount.replace('₹', '').replace(',', '')
        try:
            return float(cleaned_amount) * exchange_rates[year]
        except ValueError:
            return np.nan
    elif '$' in amount:
        # Remove commas and $ symbol
        cleaned_amount = amount.replace('$', '').replace(',', '')
        try:
            return float(cleaned_amount)
        except ValueError:
            return np.nan
    else:
        # Remove commas and convert to float if possible
        cleaned_amount = amount.replace(',', '')
        try:
            return float(cleaned_amount)
        except ValueError:
            return np.nan


#### Apply the conversion function to the column

In [186]:
combined21_18["Fund_Amount"] = combined21_18.apply(convert_currency, axis=1)

In [189]:
combined21_18.sample(n=30)

CompanyName                               Sector         Stage  \
1671            upGrad                             [EdTech]       Unknown   
1383         DailyHunt                              [Media]       Unknown   
2490          LegalPay                            [FinTech]          Seed   
234         HealthPlix     [Fitness, Health Care, Wellness]      Series A   
2616        Smartstaff                        [Recruitment]       Unknown   
1365           AsknBid                    [Investment Tech]       Unknown   
897        Genrobotics                         [AI startup]       Unknown   
1187              Crio                             [Edtech]       Unknown   
949            Dream11                      [Entertainment]       Unknown   
1621  Capital Quotient                            [Fintech]       Unknown   
1051            Eloelo                    [Social commerce]          Seed   
1305             Tring                              [Media]       Unknown   
1886         PayGlocal                 [Financial Services]      Series A   
2755         ePayLater                 [Financial Services]      Series B   
1616            Monech                            [Fintech]    Seed Round   
572         Rein Games                         [Technology]  Pre series A   
881               Miko                         [AI startup]  Pre-series B   
1041     Lido Learning                             [EdTech]      Series B   
1779          Bewakoof                  [Apparel & Fashion]       Unknown   
781              TUBBR                       [Social Media]       Unknown   
1422          Furlenco                          [Ecommerce]       Unknown   
776        Kirana King                             [Retail]  Pre-series A   
1876          Fashinza                  [Apparel & Fashion]      Series A   
2809            Gobbly                             [Retail]  Pre-series A   
2801          TravClan                         [B2B Travel]  Pre-series A   
481         Parkwheels  [Industrial, Industrial Automation]          Seed   
725             Inntot                       [Tech company]       Unknown   
1363        Convosight                     [Data Analytics]  Pre series A   
2089       Karkhana.io              [Manufacturing startup]          Seed   
2607         Rentomojo                   [Furniture Rental]       Unknown   

      Fund_Amount                   HeadQuarter  \
1671  120000000.0                        Mumbai   
1383   24000000.0                           NaN   
2490          NaN                     New Delhi   
234     3000000.0   Bangalore, Karnataka, India   
2616    4300000.0                     Bangalore   
1365          NaN                     Bangalore   
897      300000.0     Trivandrum, Kerala, India   
1187     934000.0                     Bangalore   
949   225000000.0                        Mumbai   
1621     600000.0                     Bangalore   
1051     450000.0                     Bangalore   
1305          NaN                        Mumbai   
1886    4900000.0                     Bangalore   
2755   10000000.0                        Mumbai   
1616      12700.0                        Mumbai   
572    50000000.0                         Noida   
881     3000000.0                        Mumbai   
1041    7500000.0                        Mumbai   
1779    8000000.0                        Mumbai   
781           NaN  Frisco, Texas, United States   
1422   10000000.0                     Bangalore   
776      900000.0              Jaipur, Rajastan   
1876   20000000.0                      Gurugram   
2809     900000.0                      Gurugram   
2801    2200000.0                     New Delhi   
481      175200.0       Haryana, Haryana, India   
725           NaN                        Kerala   
1363    4000000.0                     Hyderabad   
2089    1500000.0                        Mumbai   
2607   10000000.0                     Bangalore   

                                         

#### Now we fill all NaN with the median 

In [192]:
# Calculate the median of the 'Fund_Amount' column
median_value = combined21_18["Fund_Amount"].median()

In [193]:
median_value

3000000.0

In [195]:
# Fill NaN values in the "Fund_Amount" column with the median
combined21_18["Fund_Amount"].fillna(median_value, inplace=True)

In [200]:
combined21_18["Fund_Amount"].isna().any()

False

In [201]:
combined21_18.sample(n=25)

CompanyName  \
827                     Jumbotail   
123                 FeedMyPockets   
715                    Credgenics   
2325                   saarthi.ai   
985                   Hire Me Car   
1271                      YoloBus   
587                 Stanza Living   
993                    StayQrious   
558                      Pumpkart   
2203                   Slang Labs   
463                       Licious   
183         Pandorum Technologies   
2638                  Rebel Foods   
1290                        Dunzo   
234                    HealthPlix   
1085                    Mindhouse   
823                     mCaffeine   
2455                 Ecom Express   
2335                       BYJU’S   
458                   YourOwnROOM   
1998                      WizKlub   
2093                 Infra.Market   
193   Aviom India Housing Finance   
2403      Anar Business Community   
2666                    FloCareer   

                                                 Sector         Stage  \
827                                            [Retail]     Series B2   
123           [Advertising, Human Resources, Marketing]          Seed   
715                                           [FinTech]  Pre-series A   
2325                                       [AI startup]          Seed   
985                                        [TravelTech]          Seed   
1271                               [Mobility/Transport]      Series A   
587                                      [Accomodation]       Unknown   
993                                            [EdTech]          Seed   
558                                     [E-marketplace]       Unknown   
2203                                [Computer software]       Unknown   
463   [Delivery Service, Food and Beverage, Food Del...      Series C   
183                                     [Biotechnology]      Series A   
2638                                    [Cloud kitchen]      Series F   
1290                                        [Logistics]       Unknown   
234                    [Fitness, Health Care, Wellness]      Series A   
1085                                       [Healthtech]       Unknown   
823                                         [Cosmetics]      Series B   
2455                                       [E-commerce]       Unknown   
2335                                           [EdTech]       Unknown   
458       [Online Portals, Property Management, Rental]          Seed   
1998                                           [EdTech]  Pre-series A   
2093                                       [E-commerce]          Debt   
193                                           [Finance]      Series A   
2403                                     [Tech Startup]       Unknown   
2666                             [Information Services]      Series A   

      Fund_Amount                       HeadQuarter  \
827    11000000.0                         Bangalore   
123      642400.0  Bangalore City, Karnataka, India   
715     3000000.0                         New Delhi   
2325    3000000.0                         Bangalore   
985     3000000.0                             Noida   
1271    3300000.0                          Gurugram   
587     5700000.0                             Delhi   
993     2000000.0                         Bangalore   
558     3000000.0                        Chandigarh   
2203    3000000.0                         Bangalore   
463    25000000.0       Bengaluru, Karnataka, India   
183     3600000.0       Bengaluru, Karnataka, India   
2638  175000000.0                            Mumbai   
1290   55000000.0                         Bangalore   
234     3000000.0       Bangalore, Karnataka, India   
1085     286000.0                          Gurugram   
823     5000000.0                            Mumbai   
2455   20000000.0                         New Delhi   
2335   50000000.0                         Bangalore   
458      750000.0       Bangalore, Karnataka, India   
1998 